### PR TITLE
tighten up linting: add support for half-stars

### PR DESCRIPTION
**Pull request by toddmedema**
Closes https://github.com/ExpeditionRPG/expedition-app/issues/374

### CODE REVIEW

File: app/Globals.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ declare var ga: any;
 const PACKAGE = require('../package.json');
 
 export interface ReactDocument extends Document {
-  addEventListener: (e: string, f: ()=>any, useCapture?: boolean) => void;
```
Comment (smartin015) [inline app/Globals.tsx]: Why did this change?

File: app/Theme.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {
 import {fade} from 'material-ui/utils/colorManipulator'
 import spacing from 'material-ui/styles/spacing'
 
-let theme: any = {
```
Comment (smartin015) [inline app/Theme.tsx]: `export default theme: any = ...` ?

Comment (toddmedema) [inline app/Theme.tsx]: Tried that, but typescript thoroughly rejects it:
```
ERROR in [at-loader] ./app/Theme.tsx:11:21 
    TS1005: ';' expected.

ERROR in [at-loader] ./app/Theme.tsx:11:16 
    TS2304: Cannot find name 'theme'.

ERROR in [at-loader] ./app/Theme.tsx:11:23 
    TS2693: 'any' only refers to a type, but is being used as a value here.
```

File: app/Globals.tsx (hunk 1 above)
Comment (toddmedema) [inline app/Globals.tsx]: Typescript complained about it. Did some more digging and was able to tighten the definition up

**Comment by smartin015** [comment]
Is this ready for review again? Probably needs a merge :(

**Comment by toddmedema** [comment]
@smartin015 fixed fixed

**Comment by toddmedema** [comment]
Added a quick fix for https://github.com/ExpeditionRPG/expedition-app/issues/338 / showing half star ratings since it was a 2-line change / didn't want to mess with merging with linting changes going on

File: app/React.test.tsx [hunk 3]
```diff
@@ -31,21 +31,22 @@ function dummyDOM(): Document {
...
-    auth2: {
-      init: (settings: any) => {
-        return {
-          then: (fn: () => any) => {fn();}
-        };
-      },
-    },
-  };
-}
+// Unused
```
Comment (smartin015) [inline app/React.test.tsx]: Could probably just remove this.

File: app/parser/Node.tsx [hunk 4]
```diff
@@ -14,30 +14,31 @@ function getNodeAttributes(e: Cheerio): {[key:string]:string;} {
...
-
-  if (domElement === null) {
-    return selector;
-  } else if (domElement.tagName.toLowerCase() === 'quest') {
-    return 'quest ' + selector;
-  } else {
-    return '#' + domElement.attribs['id'] + selector;
-  }
-}
+// Unused
```
Comment (smartin015) [inline app/parser/Node.tsx]: Same, we can just rm.

**Pull request merged**